### PR TITLE
Allow all "Editors" to get dynamic tab styling.

### PR DESCRIPTION
**Pull request by jessegrosjean**
Allow other "Editor"s to match the special dynamic table styling that's currently given to data-type=TextEditor.

https://discuss.atom.io/t/how-to-make-my-tabs-look-like-texteditor-tabs-in-the-new-one-themes/14995/4

### CODE REVIEW

**Comment by simurai** [comment]
@jessegrosjean If I understood the problem right, your package looks like it's a TextEditor, but because the `data-type` attribute on the tab is something else, the tab doesn't look seamless?

Hmm.. I'm not exactly sure where the `data-type` comes from, I think it's whatever your package's main class is called? If so, couldn't you also call it `TextEditor` so that it gets the same type? Or would that not make sense?

**Comment by jessegrosjean** [comment]
@simurai 

> If I understood the problem right, your package looks like it's a TextEditor, but because the data-type attribute on the tab is something else, the tab doesn't look seamless?

Sorta… I would rephrase as "my package is styled liked a TextEditor (i.e. uses syntax variables for coloring), but because the data-type attribute on the tab is something else, the tab doesn't look seamless?

You are correct, one solution would be to just rename my class to "TextEditor". But I think that's not ideal, because then the name isn't really describing what it is. My package is an "Outline Editor":

https://atom.io/packages/foldingtext-for-atom

It has a public API all describing it as an outline editor, renaming the main editor component to "TextEditor" would confuse things.

Also, going forward I can see other packages having this same request. If you look at tools like Eclipse, or Xcode they support multiple editor types. For example in Xcode I can open a file as:

- Source Code Editor
- HEX Editor
- Property List Editor
- CSV Table Editor (not in Xcode, but maybe useful idea for Atom)

Assuming Atom eventually follows in the same direction (supporting multiple specialized editors) I think users would want them to be consistently styled. Right now editor authors can do that by reusing the syntax variables, the only inconsistency is that it's impossible to get the tab styling to match Atom's native text editor tab. That's what this pull request is trying to solve.

**Comment by simurai** [comment]
Thanks for explaining. I see. It would technically be possible, it just doesn't make sense naming wise.

Ok, I think your solution with this PR is great. :+1: It might add a small performance hit with the looser selector, but probably not to worry about.

**Comment by simurai** [comment]
Also changed in One light.

/cc @braver You might be interested in this too.

**Comment by jessegrosjean** [comment]
@simurai Thanks!
